hikari/impl/rest.py

```diff
@@ -431,15 +488,21 @@ def __init__(
         self._http_settings = http_settings
         self._proxy_settings = proxy_settings
 
-        self._token: typing.Union[str, rest_api.TokenStrategy, None]
+        self._token: typing.Union[str, rest_api.TokenStrategy, None] = None
+        self._token_type: typing.Optional[str] = None
```

Comment:
Are we sure we want to allow passing an empty token type?

Reply:
This only checks that token_type is or isn't None in certain cases to ensure that they passed a token type, these checks aren't here to make sure that they pass what we think is a correct token type for their scenario and if we really wanted to verify they were doing that we'd be checking against specific constants but that's not as flexible and may not account for scenarios we're not aware of down the line

Reply:
Sure ig. I just dont think we should allow not passing the type of the token

Reply:
Weird point but fun fact if they passed prefix + the whole token in as token prefix then passed " " or "" for the actual token discord would accept that as valid auth